crates/router/src/connector/utils.rs

```diff
@@ -352,6 +370,102 @@ pub fn get_header_key_value<'a>(
         ))?
 }
 
+pub fn to_boolean(string: String) -> bool {
+    let str = string.as_str();
+    match str {
+        "true" => true,
+        "false" => false,
+        "yes" => true,
+        "no" => false,
+        _ => false,
+    }
+}
+
+pub fn get_connector_meta(
+    connector_meta: Option<serde_json::Value>,
+) -> Result<serde_json::Value, Error> {
+    connector_meta.ok_or_else(missing_field_err("connector_meta_data"))
+}
+
+pub fn to_connector_meta<T>(connector_meta: Option<serde_json::Value>) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let json = connector_meta.ok_or_else(missing_field_err("connector_meta_data"))?;
+    parse_struct(json)
+}
+
+pub fn to_connector_meta_from_secret<T>(
+    connector_meta: Option<Secret<serde_json::Value>>,
+) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let connector_meta_secret =
+        connector_meta.ok_or_else(missing_field_err("connector_meta_data"))?;
+    let json = connector_meta_secret.peek().clone();
+    parse_struct(json)
+}
+
+pub fn parse_struct<T>(json: serde_json::Value) -> Result<T, Error>
```

Comment:
already there is a util function, why we need a another one?

Reply:
we need this to handle connector error, as we duplication connector error handling in all the places it will solve the problem

Reply:
Refer `crates/common_utils/src/errors.rs`, there is trait called `ErrorSwitch<T>` which needs to implemented as `ErrorSwitch<ConnectorError> for ParsingError`. Then all utils functions with `ParsingError` can be used inside `ConnectorError` context also.

Eg: json.parse_struct("typename").switch()